tools/osbuilder/rootfs-builder/rootfs.sh

```diff
@@ -671,23 +671,11 @@ EOF
 		info "Install attestation-agent with KBC ${AA_KBC}"
 		#git clone "${attestation_agent_url}" --branch "${attestation_agent_tag}" --single-branch
 		git clone --depth=1 "${attestation_agent_url}" attestation-agent
-		pushd attestation-agent/app
+
+		pushd attestation-agent
 		git fetch --depth=1 origin "${attestation_agent_version}"
 		git checkout FETCH_HEAD
-		source "${HOME}/.cargo/env"
-		target="${ARCH}-unknown-linux-${LIBC}"
-		if [ "${AA_KBC}" == "eaa_kbc" ] && [ "${ARCH}" == "x86_64" ]; then
-			RUSTFLAGS="-C link-args=-Wl,-rpath,/usr/local/lib/rats-tls"
-			# Currently eaa_kbc module only support this specific platform
-			target="x86_64-unknown-linux-gnu"
-		fi
-		if [ "$(uname -m)" != "$ARCH" ]; then
-			RUSTFLAGS+=" -C linker=$CC"
-		fi
-		export RUSTFLAGS
-		# Foreign CC is incompatible with libgit2 -- CC is still handled by `-C linker=...` flag
-		CC= cargo build --release --target "${target}" --no-default-features --features "${AA_KBC}"
-		install -D -o root -g root -m 0755 "target/${target}/release/attestation-agent" -t "${ROOTFS_DIR}/usr/local/bin/"
+		make KBC=${AA_KBC} && make install DESTDIR="${ROOTFS_DIR}/usr/local/bin/"
```

Comment:
This `make` command will build underlying crypto suites using `rust-crypto`. I wonder whether we need a condition where `openssl` is needed?

Reply:
Thanks for the comment @Xynnn007 - what are the circumstances that we'd choose to use `openssl` rather than `rust-crypto`? Is is just architecture specific e.g. `s390x` always uses `open-ssl`, or is there any other circumstances?

Reply:
I'm not sure about this. What I've done is just divided this into two suites (rust-crypto and openssl) because I saw that offline & sev kbcs were using `openssl`, and the others were using rust crypto. Now every kbc can support both crypto suites. 

So back to the question, I think it depends on the designer of offline & sev kbcs. Maybe by default we can set those KBCs use openssl, and others use rustcrypto? Or by default all use rust-crypto, and an extra parameter to enable openssl?

cc @fitzthum

Reply:
Ok, so it might be KBC specific. If they are compatible then hopefully we could avoid setting `OPENSSL=1` here and just use `rust-crypto` for decryption and then aim to move all KBCs to use that over time?

Sorry if I've missed the point here, I'm definitely not an expert on this!

Reply:
From my own point of view, yes and I hope so. I'm working on compatible issue https://github.com/confidential-containers/attestation-agent/pull/132.  However the original code of offline & sev kbcs told that `openssl` is preferred when on `s390x` to accelerate. Thus I'm not sure which option is better. Could you please give some comments abot the choices below? @fitzthum  @liudalibj 

- Set offline & sev KBCs use openssl by default, and others use rustcrypto. Or
- By default all use rust-crypto, and an extra parameter in the script to enable openssl instead

Reply:
cc @BbolroC - this discussion might be interesting to you

Reply:
My understanding was that openssl was used for the online KBS for Z compatibility reasons and for the SEV KBCs out of convenience and because there was at one point a plan to converge them with the Z stuff. Now that @Xynnn007 has extracted the crypto stuff from each KBC implementation, I"m not sure it really matters, particularly for the SEV KBCs.

Reply:
@stevenhorsman from an `s390x` perspective, `openssl` has on-ship accelerator (CPACF) support where `rust-crypto` is portable software solution. Also from a Linux distribution point of view, using `openssl` have some benefits for FIPS certification. So I would be in favor of `openssl`.

Reply:
Thanks for the info Hendrik! @Xynnn007 - I guess my question here is whether we should make the change in the attestation-agent's Makefile to switch to open-ssl for `s390x` with something like:
```
ifeq ($(ARCH), s390x)
  OPENSSL=1
endif
```
rather than passing it in here based on architecture, so then any users of the attestation-agent (including in peer pods) will automatically benefit?

Reply:
@stevenhorsman Sounds good! Let me do this -

Reply:
Sounds good! Let me handle this

Reply:
@stevenhorsman Have fixed in the https://github.com/confidential-containers/attestation-agent/pull/135/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R30

Reply:
Great - thanks! I think we can get this PR is first - once the tests are passing and then I'll add a separate PR to double check that the latest version of AA is still working before we start changing tags for the release process